packages/e2e-tests/specs/editor/various/autosave.test.js

```diff
@@ -17,13 +17,22 @@ const AUTOSAVE_INTERVAL_SECONDS = 5;
 const AUTOSAVE_NOTICE_REMOTE = 'There is an autosave of this post that is more recent than the version below.';
 const AUTOSAVE_NOTICE_LOCAL = 'The backup of this post in your browser is different from the version below.';
 
+// Save and wait for "Saved" to confirm save complete. Preserves focus in the
+// editing area.
 async function saveDraftWithKeyboard() {
-	return pressKeyWithModifier( 'primary', 's' );
+	await page.waitForSelector( '.editor-post-save-draft' );
```

Comment:
FYI, this line here was the missing key to make the tests work in a headless setting.

Reply:
Does `page.waitForSelector( '.editor-post-saved-state.is-saved' )` not check for this as well?

Reply:
They're different things:
1. Make sure that the editor is "ready" for a save. In real usage, or when testing with `PUPPETEER_HEADLESS=false` or `_SLOWMO=50`, this isn't an issue, but when testing fast it's possible to reach `saveDraftWithKeyboard` before the UI has updated to reflect its dirty state.
2. Thanks to this wait, we ensure that the <kbd>⌘S</kbd> keypress actually triggers a save, and that's when we concert the next two promises under `Promise.all`.

Reply:
it's a common issue, that we don't wait for UI elements to show up. We should do it more often to avoid false negatives.